kubeflow/trainer/utils/utils.py

```diff
@@ -269,9 +269,8 @@ def get_script_for_python_packages(
     # first url will be the index-url.
     options = [f"--index-url {pip_index_urls[0]}"]
     options.extend(f"--extra-index-url {extra_index_url}" for extra_index_url in pip_index_urls[1:])
-    # For the OpenMPI, the packages must be installed for the mpiuser.
-    if is_mpi:
-        options.append("--user")
+    # Always install for the user to avoid permission issues across environments.
+    options.append("--user")
```

Comment:
This can be removed since you always add --user to the constants, right ?
```suggestion
```
https://github.com/kubeflow/sdk/blob/b74f60e12f5841a4f099c5af547dc234fe73ea73/kubeflow/trainer/backends/localprocess/constants.py#L45

Reply:
Oh, you are right, we don't use it for k8s.
@briangallagher please rebase this PR, so we can merge it.